### PR TITLE
Chore/introduce event invitation

### LINKED ISSUE #14: As an event owner I want to send invitations by email so that others users will join my event.

### PR DESCRIPTION
**Pull request by d2bit**
I open it, but it has still some todo:
- Style the email
- Set subject
- Change the accept invitation url (it should be valid on the JS client)

It closes #14

### CODE REVIEW

File: test/mailers/previews/registrations_preview.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Preview all emails at http://localhost:3000/rails/mailers/registrations
+class RegistrationsPreview < ActionMailer::Preview
```
Comment (gponsu) [inline test/mailers/previews/registrations_preview.rb]: previews on a API?

File: app/views/v1/invitations/create.json.jbuilder [hunk 2]
```diff
@@ -0,0 +1,3 @@
+json.success true
+json.event @event.id.to_s
+json.invitationNumber @invitations.size
```
Comment (gponsu) [inline app/views/v1/invitations/create.json.jbuilder:3]: Why I want to know the number of invitations?

File: test/mailers/previews/registrations_preview.rb (hunk 1 above)
Comment (d2bit) [inline test/mailers/previews/registrations_preview.rb]: I like it to see how emails will look.
faster than sending it.

File: app/views/v1/invitations/create.json.jbuilder (hunk 2 above)
Comment (d2bit) [inline app/views/v1/invitations/create.json.jbuilder:3]: just to be sure it has created invitations for every email in the
email_list.
although.. now I'm thinking it should remove duplicated email

File: test/mailers/previews/registrations_preview.rb (hunk 1 above)
Comment (gponsu) [inline test/mailers/previews/registrations_preview.rb]: Ok, feature for development, perfect! :)

File: test/models/invitation_test.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+require 'test_helper'
+
+class InvitationTest < ActiveSupport::TestCase
+  def setup
+    @event = Event.create(name: "Moe's bar", date: Time.zone.now)
+    3.times { |number| Invitation.create(event: @event, email: "new_user#{ number }@example.com") }
```
Comment (scruti) [inline test/models/invitation_test.rb]: This is creating 3 invitations before each test. But is just used on "Delete casade" test... I would move there. 
And 3 invitations?  I would iterate less as possible (2 if you want to check multiple invitations for same event).

**Comment by scruti** [comment]
:+1:

**Pull request merged**